File: codes/easy/58 Length of Last Word/discussion.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    let count = 0
    for (let i = s.length - 1; i >= 0; i--) {
        if(s[i] === ' ') {
            if(count > 0) break;
            else continue
        } else {
            count++
        }
    }   

    return count
}

File: codes/medium/55 Jump Game/discussion.java
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let n = nums.length;
    let last_pos = n - 1;

    for(let i = n - 1; i >= 0; --i) {
        if(nums[i] + i >= last_pos) last_pos = i;
    }

    return last_pos == 0;
};